### PR TITLE
feat: basic functionality toggle for trending

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

When basic functionality toggle is on, user should only be able to search within sites.

## **Changelog**

CHANGELOG entry: User should be prompted to enable BFT when its off

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

### **After**

https://github.com/user-attachments/assets/15c342bb-40a0-4e77-8dff-5f33c88c5bbe

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Explore now respects the basic functionality toggle: hides content when off, shows an enable-CTA empty state, and limits search UI to sites.
> 
> - **Trending/Explore**:
>   - **Toggle support**: Use `selectBasicFunctionalityEnabled` to control Explore rendering and search behavior.
>     - When off:
>       - Show `BasicFunctionalityEmptyState` with CTA navigating to `Routes.SHEET.BASIC_FUNCTIONALITY`.
>       - Hide Explore home sections (`HOME_SECTIONS_ARRAY`).
>       - In search results (`ExploreSearchResults`), filter out all sections.
>       - In search bar (`ExploreSearchBar`), switch placeholder to `strings('trending.search_sites')`.
>     - When on: existing Explore content and multi-section search remain.
> - **i18n**: Add `trending.enable_basic_functionality`, `trending.basic_functionality_disabled_title`, and `trending.basic_functionality_disabled_description`.
> - **Tests**: Expand unit tests for toggle behavior across `TrendingView`, `ExploreSearchBar`, and `ExploreSearchResults`; add tests for new empty state component and navigation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2998dd5e6951ecaa9d2aa8b143a8da6013584480. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: locales/languages/en.json [hunk 1]
```diff
@@ -6997,12 +6997,16 @@
     "low_to_high": "Low to high",
     "apply": "Apply",
     "search_placeholder": "Search tokens, sites, URLs",
+    "search_sites_placeholder": "Search sites",
```
Comment (juanmigdr) [inline locales/languages/en.json]: duplicate `search_sites_placeholder` and `search_sites` lets keep only `search_sites`?

File: app/components/Views/TrendingView/TrendingView.tsx [hunk 2]
```diff
@@ -104,6 +111,38 @@ const TrendingFeed: React.FC = () => {
     });
   }, [navigation, portfolioUrl.href]);
 
+  const handleEnableBasicFunctionality = useCallback(() => {
+    navigation.navigate(Routes.MODAL.ROOT_MODAL_FLOW, {
+      screen: Routes.SHEET.BASIC_FUNCTIONALITY,
+    });
+  }, [navigation]);
+
+  const renderEmptyState = () => (
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/TrendingView.tsx]: Could we move this inside `Views/TrendingView/components`?

File: app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx [hunk 3]
```diff
@@ -78,7 +83,12 @@ const ExploreSearchResults: React.FC<ExploreSearchResultsProps> = ({
   const flatData = useMemo(() => {
     const result: FlatListItem[] = [];
 
-    SECTIONS_ARRAY.forEach((section) => {
+    // Filter sections based on basic functionality toggle
+    const sectionsToShow = isBasicFunctionalityEnabled
+      ? SECTIONS_ARRAY
+      : SECTIONS_ARRAY.filter((section) => section.id === 'sites');
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: Instead of doing filtering whcih is expensive we can just do `SECTIONS_ARRAY['sites']`

Comment (juanmigdr) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: ALSO: I am thinking the following... If we are going to show the Sites section in the search... Why shouldnt it be displayed in the main trending screen? 

I believe we should:
- Hide sites also from the search sections if functionality is off
- Show sites on the main trending screen as well as on the search

Might be one for design team

Comment (Prithpal-Sooriya) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: Is SECTIONS_ARRAY indexable via 'sites'? But good callout if we do have similar object shape instead of an array shape.

File: app/components/Views/TrendingView/TrendingView.tsx (hunk 2 above)
Comment (Prithpal-Sooriya) [reply app/components/Views/TrendingView/TrendingView.tsx]: I don't usually subscribe to single file components, but we can decouple this and convert from `renderEmptyState()` renderfn to a `<EmptyStateBasicFunc />` type of component?

Comment (Prithpal-Sooriya) [reply app/components/Views/TrendingView/TrendingView.tsx]: But if SFCs are what we are doing for this proj, we can move it to a separate file :)

File: app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx (hunk 3 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: I guess doing that wont work? SECTIONS_ARRAY[0] returns tokens section
SECTIONS_ARRAY['sites'] would return undefined ; will do  [SECTIONS_CONFIG.sites];

Comment (sahar-fehri) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: [d6292ce](https://github.com/MetaMask/metamask-mobile/pull/23252/commits/d6292ce877ef9cc62322d3c0ab717d97d6f5c19b)

File: app/components/Views/TrendingView/TrendingView.tsx (hunk 2 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/TrendingView.tsx]: [d6292ce](https://github.com/MetaMask/metamask-mobile/pull/23252/commits/d6292ce877ef9cc62322d3c0ab717d97d6f5c19b)

File: locales/languages/en.json (hunk 1 above)
Comment (sahar-fehri) [reply locales/languages/en.json]: [d6292ce](https://github.com/MetaMask/metamask-mobile/pull/23252/commits/d6292ce877ef9cc62322d3c0ab717d97d6f5c19b)

File: app/components/Views/TrendingView/TrendingView.tsx (hunk 2 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/TrendingView.tsx]: Oh i just saw Prith's comment! already moved to Views/TrendingView/components

File: app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx (hunk 3 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: sorry also i just saw this discussion lol

Comment (sahar-fehri) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: Hmmm yeah good point, do sites require api call? i thought we get them from a static list, in that case its fine to show them in search; 
but also yeah if possible to search in sites they should perhaps see it in the screen, but that kinda messes up the "empty screen" IMO XD

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { default } from './BasicFunctionalityEmptyState';
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/index.ts]: Would it be possible to move way frot hese index files? We are not using them on trending and makes things easier to navigate through? `export default xyz`

File: app/components/Views/TrendingView/TrendingView.tsx [hunk 5]
```diff
@@ -104,6 +110,12 @@ const TrendingFeed: React.FC = () => {
     });
   }, [navigation, portfolioUrl.href]);
 
+  const handleEnableBasicFunctionality = useCallback(() => {
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/TrendingView.tsx]: You can move this withing the BasicFunctionality empty component

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { strings } from '../../../../../../locales/i18n';
+
+interface BasicFunctionalityEmptyStateProps {
+  onEnablePress: () => void;
+}
+
+const BasicFunctionalityEmptyState: React.FC<
+  BasicFunctionalityEmptyStateProps
+> = ({ onEnablePress }) => (
+  <Box twClassName="flex-col w-[393px] pt-9 pb-24 justify-center items-center gap-3 flex-1">
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: Is the width necessary here? cant we remove it?

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    <Box twClassName="flex-col w-[337px] items-stretch">
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: same here

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      </Text>
+      <Text
+        variant={TextVariant.BodyMd}
+        twClassName="text-alternative text-center self-stretch font-medium"
+      >
+        {strings('trending.basic_functionality_disabled_description')}
+      </Text>
+      <Button
+        variant={ButtonVariant.Primary}
+        twClassName="h-[48px] self-stretch mt-6"
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: same here for height (if poss)

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.test.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import BasicFunctionalityEmptyState from './BasicFunctionalityEmptyState';
+
+describe('BasicFunctionalityEmptyState', () => {
+  it('renders title text', () => {
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.test.tsx]: Te first 3 test cases can be combined into 1 and just say "renders empty state" and checks for all props instead of rendering 3 separate times

File: app/components/Views/TrendingView/TrendingView.test.tsx [hunk 10]
```diff
@@ -588,4 +594,52 @@ describe('TrendingView', () => {
 
     expect(mockNavigate).toHaveBeenCalledWith('ExploreSearch');
   });
+
+  describe('basic functionality toggle', () => {
+    it('displays empty state when basic functionality is disabled', () => {
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/TrendingView.test.tsx:446]: These 3 props are already checked within the component:
```
expect(getByText('Explore is not available')).toBeDefined();
      expect(
        getByText(
          "We can't fetch the required metadata when basic functionality is disabled.",
        ),
      ).toBeDefined();
      expect(getByText('Enable basic functionality')).toBeDefined();
```

Worth adding a testId to emptystate component and just checking its displayed

File: app/components/Views/TrendingView/TrendingView.test.tsx [hunk 11]
```diff
@@ -588,4 +594,52 @@ describe('TrendingView', () => {
...
+      expect(getByText('Explore is not available')).toBeDefined();
+      expect(
+        getByText(
+          "We can't fetch the required metadata when basic functionality is disabled.",
+        ),
+      ).toBeDefined();
+      expect(getByText('Enable basic functionality')).toBeDefined();
+    });
+
+    it('navigates to bottom sheet when enable button is pressed', () => {
```
Comment (juanmigdr) [inline app/components/Views/TrendingView/TrendingView.test.tsx]: This case can be moved to the component itself once te functionality is moved there

File: app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx (hunk 3 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/ExploreSearchScreen/components/ExploreSearchResults/ExploreSearchResults.tsx]: ALright i think we can remove sites all together since we get them from an api call [https://github.com/MetaMask/metamask-mobile/blob/0fcf5c979f4ac1f92c16dd01a801af4c3[…]omponents/Views/TrendingView/SectionSites/hooks/useSitesData.ts](https://github.com/MetaMask/metamask-mobile/blob/0fcf5c979f4ac1f92c16dd01a801af4c3f9f0484/app/components/Views/TrendingView/SectionSites/hooks/useSitesData.ts#L68)

will wait for Eric's confirmation

**Review by Prithpal-Sooriya — DISMISSED** [review]

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/index.ts (hunk 4 above)
Comment (Prithpal-Sooriya) [reply app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/index.ts]: hehehe, not sure why we need index/barrel files. No strong opinions on this (I do sway for direct imports instead of through barrel files).

But this can be done in a refactor.

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx (hunk 6 above)
Comment (Prithpal-Sooriya) [reply app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: btw, if we find that we need some custom spacing or sizing, lets chat with Erik to confirm -- usually it it best to stick with the standard spaces and sizing

Comment (sahar-fehri) [reply app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: Hmm good callout this one is not necessary

File: app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx (hunk 7 above)
Comment (sahar-fehri) [reply app/components/Views/TrendingView/components/BasicFunctionalityEmptyState/BasicFunctionalityEmptyState.tsx]: without the width in here it would make the container stretch more than it should 🤔   <Box twClassName="flex-col w-[337px] items-stretch">

**Review by juanmigdr — DISMISSED** [review]
LGTM

**Review by juanmigdr — APPROVED** [review]
